modules/core/test/src/smithy4s/ShapeIdHintsSmokeSpec.scala

```diff
@@ -0,0 +1,152 @@
+package smithy4s
+
+import java.util.UUID
+import schematic.ByteArray
+import schematic.Field
+import schematic.Alt
+import cats.Id
+import schematic.struct.GenericAritySchematic
+import smithy4s.capability.Invariant
+
+object ShapeIdHintsSmokeSpec extends weaver.FunSuite {
+
+  type HintTracker[A] = TestHinted[Id, A]
+
+  case class TestHinted[F[_], A](hints: List[Hint], make: List[Hint] => F[A]) {
+    def addHints(h: List[Hint]): TestHinted[F, A] = copy(hints = hints ++ h)
+    def imap[B](to: A => B, from: B => A)(implicit
+        I: Invariant[F]
+    ): TestHinted[F, B] =
+      TestHinted(hints, h => I.imap(make(h))(to, from))
+  }
+
+  object TestHinted {
+    def apply[F[_]]: PartiallyAppliedHintedTest[F] =
+      new PartiallyAppliedHintedTest[F]
+
+    class PartiallyAppliedHintedTest[F[_]] {
+      def from[A](f: List[Hint] => F[A]): TestHinted[F, A] =
+        TestHinted(List.empty, f(_))
+    }
+  }
+
+  implicit def invariantFromCats[F[_]: cats.Invariant]: Invariant[F] =
+    new Invariant[F] {
+      def imap[A, B](fa: F[A])(to: A => B, from: B => A): F[B] =
+        cats.Invariant[F].imap(fa)(to)(from)
+    }
+
+  private object TestCompiler
```

Comment:
First thing I think seeing this...

![image](https://user-images.githubusercontent.com/894884/149059295-2e2db4af-3497-4a1e-a92f-7a120003c567.png)


I feel like this could be much simpler, a `StubSchematic` that compiles everything to its members' shape IDs. Unless we need the actual values provided by schematic 😅 

This seems to pass the two tests you already had passing:

```scala
type ToShapeIds[A] = List[ShapeId]

object SimpleTestCompiler extends StubSchematic[ToShapeIds] {
  def default[A]: List[ShapeId] = Nil
  override def bijection[A, B](
      f: ToShapeIds[A],
      to: A => B,
      from: B => A
  ): ToShapeIds[B] = f

  override def genericStruct[S](fields: Vector[Field[ToShapeIds, S, _]])(
      const: Vector[Any] => S
  ): ToShapeIds[S] =
    fields.flatMap(_.instance).toList

  override def withHints[A](fa: ToShapeIds[A], hints: Hints): ToShapeIds[A] =
    fa ++ hints.get(ShapeId)
}
```

Reply:
We could just as well make this collect all the hints if we want to test for that in the future. The only concern I have right now is whether we want to check which field has which ID, which would be a bit more complicated.

Reply:
> I have right now is whether we want to check which field has which ID, which would be a bit more complicated. 

nah, I'm fine not doing it

Reply:
I like @kubukoz's idea. Add a method for unions (and corresponding shapes in the model for the test to retrieve), and I think we're good.

Reply:
Thanks for saving me from myself @kubukoz 😅 haha this looks great!